Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -798,11 +798,10 @@ const static HashSet<AtomicString>& codecSet()
             Vector<AtomicString> webkitCodecs;
         };
 
-        std::array<GstCapsWebKitMapping, 3> mapping = { {
-            { VideoDecoder, "video/x-h264,  profile=(string){ constrained-baseline, baseline }", { "x-h264" } },
-            { VideoDecoder, "video/x-h264, stream-format=avc", { "avc*"} },
+        GstCapsWebKitMapping mapping[] = {
```

Comment:
Zan originally suggested a std::array for this: https://bugs.webkit.org/show_bug.cgi?id=167120#c5

I wonder how the for loop is able to know where the array finishes. If there isn't a clear answer to that question, better leave it as an std::array.

Apart from this, the addition of "avc*" is fine. This could be upstreamed if upstream (or playbin) is also able to place an h264parse element automatically to do the conversion.

Reply:
Ok, I've found it, the compiler can know it: http://en.cppreference.com/w/cpp/language/range-for
r+ then.

Reply:
About `std::array`, what I changed it to is a fixed size array (the `[]` syntax with an initializer is a shorthand, the compiler automatically fills in the number of elements into the mapping type). Because the size of the array is fixed, the range-based loops can just go from `mapping` to `mapping + N_ELEM(mapping)`, and that's what it does.

Reply:
Oh, I see you merged it already, thanks!